x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/AuthorizationServiceTests.java

```diff
@@ -922,26 +923,26 @@ public void testServiceAccountDenial() {
         );
         assertThat(
             securityException,
-            throwableWithMessage(
-                containsString("[" + action + "] is unauthorized" + " for " + "service account" + " [" + serviceUser.principal() + "]")
-            )
+            throwableWithMessage(containsString("[" + action + "] is unauthorized for service account [" + serviceUser.principal() + "]"))
         );
-        if (isRunAs) {
-            assertThat(securityException, throwableWithMessage(containsString("run as [" + finalUser.principal() + "] with roles [")));
-        }
-        assertThat(securityException, throwableWithMessage(containsString("this action is granted by the index privileges [read,all]")));
         if (isRunAs && false == canRunAs) {
             verify(auditTrail).runAsDenied(eq(requestId), eq(authentication), eq(action), eq(request), authzInfoRoles(role.names()));
+            assertThat(securityException, throwableWithMessage(containsString("run as [" + finalUser.principal() + "]")));
```

Comment:
```suggestion
            assertThat(securityException, throwableWithMessage(containsString("is unauthorized to run as [" + finalUser.principal() + "]")));
```
Otherwise, the erorr message for unauthorized action (as opposed to denied-run-as), i.e. "run as [...] with roles [" can also match the assertion.